src/trackers/EventTracker.ts

```diff
@@ -2,33 +2,49 @@ import { IEvent } from "../events/core";
 import { GamePlayMode } from "../events/core";
 
 export default class EventTracker {
+    readonly SEASONAL_CHANCE = 30;
     private pool: IEvent[];
     private queue: IEvent[];
+    private seasonal: IEvent[][];
 
-    constructor(eventPool: IEvent[], eventQueue: IEvent[]) {
+    constructor(eventPool: IEvent[], eventQueue: IEvent[], seasonalPool: IEvent[][]) {
         this.pool = eventPool;
         this.queue = eventQueue;
+        this.seasonal = seasonalPool;
     }
 
-    public getNextEvent(): IEvent {
+    public getNextEvent(week: number): IEvent {
         // The queue being non-empty indicates there is a follow up
         // event to go through, otherwise pick a random event
         if (this.queue.length > 0)
             return this.queue.shift() as IEvent;
-        else if (this.pool.length > 0){
+
+        const r = Math.random() * 100;
+        const month = Math.floor(week / 4) + 9;

```

Comment:
why do you need to add 9? is it because of the year starts in sept?

Reply:
also, should the month be -1, since the array starts at 0?

Reply:
We start with `week = 0`, which gives us the 9th month. It is a good point though, since I haven't capped the month with a `% 12`. It also means we will have an off-by-one error whenever we add this. Good point.